MIGRATION.md

```diff
@@ -300,6 +300,32 @@ collection.findOneAndUpdate(
 
 Dates are worth a special mention. Codecs are already provided for Java time, allowing their use directly in filters/updates. However, they do not exist for jodatime, you will have to use `Codecs.toBson` with the `uk.gov.hmrc.mongo.play.json.formats.MongoJodaFormats` in scope. We recommend migrating to use java time.
 
+## UUIDs
+
+MongoDB is able to serialize UUIDs as binary values for greater storage efficiency and index performance.
+
+However, it's important to be aware that there are two different formats in which the Mongo Java driver encodes UUIDs as BSON.
+
+The first is the legacy binary subtype `0x03` UUID encoding. Values of this type will appear as `LUUID`, `Legacy UUID` or `BinData(3, "...")` in Mongo database tools like Robo 3T and Mongo Compass.
+
+The second is the new binary subtype `0x04` UUID encoding. Values of this type will appear as `UUID` or `BinData(4, "...")` in Mongo database tools like Robo 3T and Mongo Compass.
+
+There is a new binary subtype because in the past, different Mongo drivers used different binary encodings for UUIDs of subtype `0x03`, which caused problems in multi-language environments. As a result a new standardised binary subtype `0x04` for UUIDs was created, which uses a common encoding in all Mongo drivers.
+
+This means that if your service currently uses binary UUIDs and you are migrating to **hmrc-mongo**, then you must be aware of which encoding your service currently uses.
+
+If you do not configure the UUID encoding to match your existing data, queries which filter by UUID will fail to find any data that was written prior to migrating to **hmrc-mongo**. If you are forced to back out your migration to **hmrc-mongo**, any new data that was written after the deployment of the new code will not be able to be found once the changes are backed out.
```

Comment:
I think "fail to find" should be "throw a runtime exception" - can you confirm?

Reply:
Ah yes, I was thinking of this in the context of find queries but I will expand on that a bit

Reply:
It probably depends on what you are doing, could just say "fail to find or even throw a runtime exception"?
Maybe could also add "You could consider using an aggregate pipeline to migrate the data first"

Reply:
I have mentioned that they might see runtime exceptions when reading data, and that it might be practical to migrate the data first now 👍